compiler/ghc/default.nix

```diff
@@ -315,6 +322,12 @@ stdenv.mkDerivation (rec {
 
     # Used to detect non haskell-nix compilers (accidental use of nixpkgs compilers can lead to unexpected errors)
     isHaskellNixCompiler = true;
+
+    # The same GHC, but with debug enabled (if it can be)
+    dwarf = lib.makeOverridable self (args // {
```

Comment:
Perhaps stupid question, but could we just enable it by default where it's possible, rather than having a separate attribute with the DWARF-enabled compiler? Or is there a penalty to enabling it that we don't want in all cases?

Reply:
Compiler derivation is twice as large.  I suspect the same will be true of all the library components as well.  So it would double the disk usage.  Ideally we would split dwarf info out into separate derivations somehow.

Reply:
Ouch, that's pretty painful :grimacing: I just hate options :sweat_smile:

Reply:
And it would be nice if enabling debugging was a "rebuild all my deps" change rather than a "rebuild the compiler" change.

Reply:
It also doubles the number of compilers haskell.nix builds on CI.  Well it only adds 3 right now, but once we add GHC 9.0.1 and drop 8.6.5 it will be worse and if darwin and cross compiling get dwarf support it would be worse still.

Perhaps instead of building the non dwarf compiler we should just have it copy the dwarf one, use replaceDependency to replace references to itself and then strip the libraries.  I think that would get us most of the way there.  Users who use both versions would still loose out disk space and download wise, but at least it would only be an extra 50%.

Maybe we could even do the same for `comp-builder.nix`.

I guess it also means that if you build the non debug GHC locally (rather than downloading it) you will need about 6GB for it to complete.  That might suck.  Cachix's community plan for instance has a 10 GB limit (`nix-store --delete` the dwarf GHC before pushing to cachix might help with that).

Reply:
> Maybe we could even do the same for `comp-builder.nix`.

No, that would require everyone to have the dwarf version of GHC to build anything.